contracts/flow/teleport/TeleportCustody.cdc

```diff
@@ -1,176 +1,224 @@
+			pre { 
+				!TeleportCustody.isFrozen:
+					"Teleport service is frozen"
+				to.length == TeleportCustody.teleportAddressLength:
+					"Teleport address should be teleportAddressLength bytes"
+			}
+			let vault <- from as! @BloctoToken.Vault
+			let fee <- vault.withdraw(amount: self.lockFee)
+			self.feeCollector.deposit(from: <-fee)
+			let amount = vault.balance
+			TeleportCustody.lockVault.deposit(from: <-vault)
+			emit Locked(amount: amount, to: to)
+			emit FeeCollected(amount: self.lockFee, type: 0)
+		}
+		
+		access(AdminEntitlement)
+		fun unlock(amount: UFix64, from: [UInt8], txHash: String): @{FungibleToken.Vault} { 
+			pre { 
+				!TeleportCustody.isFrozen:
+					"Teleport service is frozen"
+				amount <= self.allowedAmount:
+					"Amount unlocked must be less than the allowed amount"
+				amount > self.unlockFee:
+					"Amount unlocked must be greater than unlock fee"
+				from.length == TeleportCustody.teleportAddressLength:
+					"Teleport address should be teleportAddressLength bytes"
+				txHash.length == TeleportCustody.teleportTxHashLength:
+					"Teleport tx hash should be teleportTxHashLength bytes"
+				!(TeleportCustody.unlocked[txHash] ?? false):
+					"Same unlock txHash has been executed"
+			}
+			self.allowedAmount = self.allowedAmount - amount
+			TeleportCustody.unlocked[txHash] = true
+			emit Unlocked(amount: amount, from: from, txHash: txHash)
+			let vault <- TeleportCustody.lockVault.withdraw(amount: amount)
+			let fee <- vault.withdraw(amount: self.unlockFee)
+			self.feeCollector.deposit(from: <-fee)
+			emit FeeCollected(amount: self.unlockFee, type: 1)
+			return <-vault
+		}
+		
+		access(AdminEntitlement)
+		fun withdrawFee(amount: UFix64): @{FungibleToken.Vault} { 
+			return <-self.feeCollector.withdraw(amount: amount)
+		}
+		
+		access(AdminEntitlement)
+		fun updateLockFee(fee: UFix64) { 
+			self.lockFee = fee
+		}
+		
+		access(AdminEntitlement)
+		fun updateUnlockFee(fee: UFix64) { 
+			self.unlockFee = fee
+		}
+		
+		access(AdminEntitlement)
+		fun getFeeAmount(): UFix64{  
+			return self.feeCollector.balance
+		}
+		
+		access(all)
+		fun depositAllowance(from: @Allowance) { 
+			self.allowedAmount = self.allowedAmount + from.balance
+			destroy from
+		}
+		
+		init(allowedAmount: UFix64) { 
+			self.allowedAmount = allowedAmount
+			self.feeCollector <- BloctoToken.createEmptyVault(vaultType: Type<@BloctoToken.Vault>()) as! @BloctoToken.Vault
+			self.lockFee = 3.0
+			self.unlockFee = 0.01
+		}
+	}
+	
+	access(all)
+	fun getLockVaultBalance(): UFix64 { 
+		return TeleportCustody.lockVault.balance
+	}
+	
+	init() { 
+		self.teleportAddressLength = 20
+		self.teleportTxHashLength = 64
+		self.AdminStoragePath = /storage/TeleportCustodyAdmin
+		self.TeleportAdminStoragePath = /storage/TeleportCustodyTeleportAdmin
+		self.TeleportAdminTeleportUserPath = /public/TeleportCustodyTeleportUser
+		self.TeleportAdminTeleportControlPath = /private/TeleportCustodyTeleportControl
```

Comment:
The private path is removed in Cadence 1.0. Please check the other files as well.

Reply:
but this is the init function, can we modify that and should we?